src/Sylius/Bundle/CoreBundle/OAuth/UserProvider.php

```diff
@@ -132,21 +141,34 @@ private function createUserByOAuthUserResponse(UserResponseInterface $response)
     {
         /** @var SyliusUserInterface $user */
         $user = $this->userFactory->createNew();
-        /** @var CustomerInterface $customer */
-        $customer = $this->customerFactory->createNew();
+
+        $customer = $this->customerRepository->findOneBy(['emailCanonical' => $response->getEmail()]);
```

Comment:
You should probably call `strtolower()` cause you can't expect that response will contain email in lowercase.